### ISSUE TITLE
Escape characters present in log line are being rendered

### ISSUE THREAD

**Issue by daghaian** [grafana/grafana#41120]
**What happened**: 

Currently, I'm using Grafana in conjunction with a **Loki** datasource. If the ingested JSON log messages happen to contain a newline or other escape characters (ex. {"error":"this is an error\n\t"}), then Grafana renders the newline and tab and consequently, I'm unable to apply the JSON parser as its no longer considered valid JSON.

**What you expected to happen**:

I would expect that either Grafana opts not to render the escape character or at the very least provides a toggle. If it does choose to render said characters, it should not be influencing its ability to apply the JSON parser to the results.

**How to reproduce it (as minimally and precisely as possible)**:
Log data is being ingested by fluentbit loki plugin. An example of the log returned by querying Loki can be seen below

```"{\"environment\":\"aws\",\"error\":\"This is a test error message\n\tand its no good\",\"identity\":\"logging_team\",\"level\":\"error\",\"message\":\"failed to schedule key for deletion with ARN\"}"```

As you can see, there is a newline and tab escape character in the error message at the time of query and Grafana opts to render these, leading to an inability to use the JSON parser.

**Anything else we need to know?**:

**Environment**:
- Grafana version: v8.2.2 (6232fe07c0)
- Data source type & version: Loki v2.3.0 w/ data type JSON (see example above)
- OS Grafana is installed on: Running in AWS ECS w/ Fargate
- User OS & Browser: Mac OSX 10.15.7 + Chrome 95.0.4638.54
- Grafana plugins: N/A
- Others: N/A

**Comment by zuchka** [comment]
Thanks for creating this issue, @daghaian 

There is a history of issues and PRs surrounding this issue. See:

* https://github.com/grafana/loki/issues/3245
* https://github.com/grafana/loki/pull/3240
* https://github.com/grafana/loki/issues/2562
* https://github.com/grafana/loki/pull/2286
* https://github.com/grafana/loki/issues/2246

But these are all in the Loki repo. Do think your problem is with more the visualization of logs inside Grafana / Explore, or with Loki proper? If Loki, then I'd advise closing and moving this over to the [loki repo](https://github.com/grafana/loki)

@ethirolle any input?

**Comment by daghaian** [comment]
@zuchka  Thank you for the comment. Correct me if I'm wrong, but if loki itself has ingested the log in the format I linked above:

```"{\"environment\":\"aws\",\"error\":\"This is a test error message\n\tand its no good\",\"identity\":\"logging_team\",\"level\":\"error\",\"message\":\"failed to schedule key for deletion with ARN\"}"```

would we not expect Grafana to avoid rendering the actual escape characters in the entry? My thought process is that by rendering them, the JSON is considered invalid which would not be the behavior we'd want. I spent a good long time trying to figure out where I feel the issue actually lies (loki/grafana) but it seems to be that at each stage of the pipeline leading up to Grafana's ingestion, the value is being stored as a proper string. It's only after it's been queried and displayed in Grafana that using the JSON parser seems to have a problem.
